addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4671,6 +4682,183 @@ registry.ReplaceMedia = SnippetOptionWidget.extend({
         // to be refactored when the new editor is merged
         this.$target.dblclick();
     },
+    /**
+     * Adds an anchor to the url.
+     * Here "anchor" means a specific section of a page.
+     *
+     * @see this.selectClass for parameters
+     */
+    setAnchor(previewMode, value) {
+        const linkEl = this.$target[0].parentElement;
+        let url = linkEl.getAttribute('href');
+        if (url.includes('#')) {
+            url = url.split('#')[0];
+        }
+        linkEl.setAttribute('href', url + value);
+    },
+    /**
+     * Makes the image a clickable link by wrapping it in an <a>.
+     * This function is also called for the opposite operation.
+     *
+     * @see this.selectClass for parameters
+     */
+    setLink() {
+        let linkEl = this.$target[0].parentElement;
+        const $urlInputEl = this.$urlInputEl;
+        if (linkEl.tagName !== 'A') {
+            // Wrap this.$target[0] like this: <a>this.$target[0]</a>.
+            const parentEl = this.$target[0].parentNode;
+            linkEl = document.createElement('a');
+            parentEl.replaceChild(linkEl, this.$target[0]);
+            linkEl.appendChild(this.$target[0]);
+            // Set focus on the URL field.
+            setTimeout(function () { // Wait for the URL input to appear.
+                $urlInputEl.focus();
+            }, 300);
+        } else {
+            // Remove the parent anchor.
+            this.$target[0].parentNode.replaceWith(this.$target[0]);
+            $urlInputEl.val('');
+        }
+    },
+    /**
+     * Makes that when the link (parent element of the image) is clicked, the
+     * url will open on another page. This function is also called for the
+     * opposite operation (make the url open in the current browser page).
+     *
+     * @see this.selectClass for parameters
+     */
+    setNewWindow() {
+        const linkEl = this.$target[0].parentElement;
+        if (linkEl.getAttribute('target') && linkEl.getAttribute('target') === '_blank') {
+            return linkEl.removeAttribute('target');
+        }
+        linkEl.setAttribute('target', '_blank');
+    },
+    /**
+     * Records the target url of the hyperlink.
+     *
+     * @see this.selectClass for parameters
+     */
+    setUrl() {
+        const linkEl = this.$target[0].parentElement;
+        if (linkEl.tagName !== 'A') {
+            return;
+        }
+        let url = this.$urlInputEl.val();
+        if (url && !url.startsWith('/') && !url.startsWith('#') && !url.startsWith('http')) {
+            url = 'http://' + url;
+        }
+        const oldHref = linkEl.getAttribute('href') || '';
+        linkEl.setAttribute('href', url);
+        if ((oldHref.length === 0 && url.length > 0) || (oldHref.length > 0 && url.length === 0)) {
+            this.updateUIVisibility(); // Show or hide the anchors we-select.
+        }
+        this.rerender = true;
```

Comment:
I do not know if it is related to this line, but any change to the URL triggers an heavy operation that makes the loading effect appear.
This does not happen when changing the URL in the text-link tools.